packages/doenetml-worker-rust/lib-doenetml-core/src/core/components/doenet/text_input/component.rs

```diff
@@ -2,19 +2,23 @@ use serde::{Deserialize, Serialize};
 use strum::VariantNames;
 use strum_macros::EnumVariantNames;
 
-use crate::components::prelude::*;
+use crate::{
+    components::prelude::*,
+    general_state_var::{BooleanStateVar, StringStateVar},
+};
 
 use super::TextInputState;
 
-#[derive(Debug, Deserialize, EnumVariantNames)]
-#[serde(rename_all = "camelCase")]
-#[strum(serialize_all = "camelCase")]
+#[derive(Debug, AttributeStateVar)]
 pub enum TextInputAttribute {
     /// Whether the `<textInput>` should be hidden.
+    #[component_attribute(state_var = BooleanStateVar, default = false, explicit_type = bool)]
     Hide,
     /// Whether the `<textInput>` should be editable.
+    #[component_attribute(state_var = BooleanStateVar, default = false, explicit_type = bool)]
     Disabled,
     /// The content that should prefill the `<textInput>`, giving it a default value before a user has interacted with the input.
+    #[component_attribute(state_var = StringStateVar, default = "\"\"", explicit_type = String)]
```

Comment:
I'm confused why you are using the string containing quotes as the default. I'd assume we want the default to be a blank string.

Reply:
The string containing quotes is the empty string...One layer of quotes is peeled off if present. I don't know why or how to change it...

Reply:
Would `String::new()` be easier to understand?

Reply:
Sure, I like `String::new()`, as it removes the confusion. Strange how it peels off quotes.